test/cpp/api/grad_mode.cpp

```diff
@@ -0,0 +1,86 @@
+#include <torch/script.h>
+#include <gtest/gtest.h>
+#include <test/cpp/api/support.h>
+
+using namespace torch::autograd;
+namespace {
+  torch::Tensor functional_op(torch::Tensor& x) {
+    return x + x;
+  }
+
+  void inplace_op(torch::Tensor& x) {
+    x.add_(1);
+  }
+
+  torch::Tensor view_op(torch::Tensor& x) {
+    return x.view({2, 3});
+  }
+
+  void assert_tensor_creation_meta(torch::Tensor& x, torch::autograd::CreationMeta creation_meta) {
+    ASSERT_EQ(static_cast<torch::autograd::DifferentiableViewMeta*>(x.unsafeGetTensorImpl()->autograd_meta())->get_creation_meta(), creation_meta);
```

Comment:
nit: you might wat to add a check that the autograd meta is not null and has a backward view before this no? Or we are ok with this segfaulting if we pass in a Tensor that does not verify these?